src/decode.rs

```diff
@@ -2940,6 +2955,8 @@ unsafe fn decode_b_inner(
                 TileStateRef::Frame => &f.lf.lvl,
                 TileStateRef::Local => &ts.lflvlmem,
             };
+            let mut a_uv_guard;
+            let mut l_uv_guard;
```

Comment:
Do you need these guards?

Reply:
Same as my other comment: We need the guards for the pair of slices that are being passed in as a tuple.